File: fluentcore/src/main/java/com/mcreater/fxfluent/stage/controls/FluentCloseButton.java
package com.mcreater.fxfluent.stage.controls;

import com.mcreater.fxfluent.controls.FluentButton;
import com.mcreater.fxfluent.controls.icon.FluentIcon;
import com.mcreater.fxfluent.util.listeners.NewValueListener;
import com.mcreater.fxfluent.xaml.style.SystemThemeLoop;
import javafx.scene.Node;
import javafx.scene.control.Button;
import javafx.scene.paint.Color;

import java.util.stream.Stream;

import static com.mcreater.fxfluent.controls.state.StateUtil.genState;

public class FluentCloseButton extends Button {
    private FluentIcon icon;
    public FluentCloseButton() {
        this.init();
    }

    private FluentCloseButton(String var1) {
        super(var1);
        this.init();
    }

    public FluentCloseButton(String var1, Node var2) {
        super(var1, var2);
        this.init();
    }
    private void init() {
        getStyleClass().add("fluent-close-button");
        setPrefSize(48, 34);
        setMaxSize(48, 34);
        setMinSize(48, 34);
        icon = new FluentIcon('\uE8BB');
        setGraphic(icon);
        icon.getStyleClass().add("fluent-icon");
        icon.setTextFill(SystemThemeLoop.isDark() ? Color.WHITE : Color.BLACK);

        Stream.of(
                hoverProperty(),
                pressedProperty(),
                focusedProperty(),
                disabledProperty()
        ).forEach(a -> a.addListener((NewValueListener<Boolean>) t1 -> {
            switch (genState(isDisabled(), isHover(), isPressed(), isFocused())) {
                case FOCUS:
                case NONE:
                    icon.setTextFill(SystemThemeLoop.isDark() ? Color.WHITE : Color.BLACK);
                    break;
                case HOVER:
                    icon.setTextFill(Color.WHITE);
                    break;
                case PRESS:
                case DISABLE:
                    icon.setTextFill(new Color(
                            1,
                            1,
                            1,
                            0.7
                    ));
                    break;
            }
        }));
    }
    public String getUserAgentStylesheet() {
        return FluentButton.class.getClassLoader().getResource(SystemThemeLoop.isDark() ? "css/FluentCloseButtonDark.css" : "css/FluentCloseButtonLight.css").toString();
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/stage/controls/FluentCloseButton.kt
package com.mcreater.fxfluent.stage.controls

import com.mcreater.fxfluent.controls.FluentButton
import com.mcreater.fxfluent.controls.icon.FluentIcon
import com.mcreater.fxfluent.controls.state.StateType
import com.mcreater.fxfluent.controls.state.StateUtil
import com.mcreater.fxfluent.util.listeners.NewValueListener
import com.mcreater.fxfluent.xaml.style.SystemThemeLoop
import javafx.beans.property.ReadOnlyBooleanProperty
import javafx.scene.control.Button
import javafx.scene.paint.Color
import java.util.stream.Stream

class FluentCloseButton: Button() {
    init {
        init()
    }
    private var icon: FluentIcon? = null
    private fun init() {
        styleClass.add("fluent-close-button")
        setPrefSize(48.0, 34.0)
        setMaxSize(48.0, 34.0)
        setMinSize(48.0, 34.0)
        icon = FluentIcon('\uE8BB')
        graphic = icon
        icon!!.styleClass.add("fluent-icon")
        icon!!.textFill = if (SystemThemeLoop.isDark()) Color.WHITE else Color.BLACK
        Stream.of(
            hoverProperty(),
            pressedProperty(),
            focusedProperty(),
            disabledProperty()
        ).forEach { a: ReadOnlyBooleanProperty ->
            a.addListener(
                NewValueListener {
                    when (StateUtil.genState(isDisabled, isHover, isPressed, isFocused)) {
                        StateType.FOCUS, StateType.NONE -> icon!!.textFill =
                            if (SystemThemeLoop.isDark()) Color.WHITE else Color.BLACK

                        StateType.HOVER -> icon!!.textFill = Color.WHITE
                        StateType.PRESS, StateType.DISABLE -> icon!!.textFill = Color(
                            1.0,
                            1.0,
                            1.0,
                            0.7
                        )
                    }
                })
        }
    }

    override fun getUserAgentStylesheet(): String {
        return FluentButton::class.java.getClassLoader()
            .getResource(if (SystemThemeLoop.isDark()) "css/FluentCloseButtonDark.css" else "css/FluentCloseButtonLight.css")
            ?.toString() ?: ""
    }
}